test/test_fake_tensor.py

```diff
@@ -212,6 +212,16 @@ def test_fake_grad_copy(self):
 
         self.assertTrue(isinstance(fake_x.grad, FakeTensor))
 
+    @unittest.skipIf(not RUN_CUDA, "requires cuda")
+    def test_index_put_error(self):
+        mode = FakeTensorMode()
+        for context in [contextlib.nullcontext, lambda: mode]:
+            with context():
+                y = torch.randn(4, 2, 2, 3)
+                x = torch.randn(2, 2, 3).to('cuda')
+                with self.assertRaises(RuntimeError):
+                    x[[1, 2]] = y
```

Comment:
this would raise error with shape mismatch even if the devices were correct

Reply:
It doesn't

Reply:
```
In [2]: y = torch.randn(4, 2, 2, 3)
   ...: x = torch.randn(2, 2, 3)

In [3]: x[[1,2]]=y
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 x[[1,2]]=y

RuntimeError: shape mismatch: value tensor of shape [4, 2, 2, 3] cannot be broadcast to indexing result of shape [2, 2, 3]
```

Reply:
well, the fake tensor version doesnt :/

Reply:
Well, that's also worrying

Reply:
We don't do a good job of erroring in FakeTensor when eager errors. This is the whole "test bankruptcy". it would be nice to fix but has generally been seen as low priority and no one is signed up to do it

Reply:
I don't really expect this to work consistently today